src/wiring_diagrams/CPortGraphs.jl

```diff
@@ -0,0 +1,194 @@
+module CPortGraphs
+using ...Theories
+using ...CategoricalAlgebra
+import ...CategoricalAlgebra.CSets: migrate!
+using ...Present
+using ...Graphs
+using ...Graphs.BasicGraphs
+import ...Graphs.BasicGraphs: TheoryGraph
+import ..DirectedWiringDiagrams: ocompose
+
+export ThCPortGraph, ThOpenCPortGraph, ThSymCPortGraph, ThOpenSymCPortGraph, ThBundledCPG,
+  CPortGraph, OpenCPortGraph, SymCPortGraph, OpenSymCPortGraph, BundledCPG
+
+
+@present ThCPortGraph(FreeSchema) begin
+    B::Ob
+    P::Ob
+    W::Ob
+
+    src::Hom(W,P)
+    tgt::Hom(W,P)
+    box::Hom(P,B)
+end
+
+@present ThOpenCPortGraph <: ThCPortGraph begin
+    OP::Ob
+    con::Hom(OP, P)
+end
+
+@present ThSymCPortGraph <: ThCPortGraph begin
+    inv::Hom(W, W)
+
+    compose(inv,inv) == id(W)
+    compose(inv,src) == tgt
+    compose(inv,tgt) == src
+end
+
+@present ThOpenSymCPortGraph <: ThSymCPortGraph begin
+    OP::Ob
+
+    con::Hom(OP, P)
+end
+
+const CPortGraph = ACSetType(ThCPortGraph)
```

Comment:
The pattern I usually follow for C-set types is:

```julia
const AbstractX = AbstractACSetType(TheoryX)
const X = ACSetType(TheoryX, index=[...])
```

Importantly, the abstract type will match any ACSet type for that theory, irrespective of indexing, whereas the concrete type only matches the type with *exactly* the given indices. Thus, the methods bellow should be for the abstract types, not the concrete ones.

For graphs, I usually follow LightGraphs in putting `index=[:src, :tgt]` for graphs and `index=[:src]` for symmetric graphs. This seems like a good default. Advanced users who want to disable indexing for specialized performance reasons can do so by making their own types.